app/src/models/searchCard.ts

```diff
@@ -20,14 +30,48 @@ export class SearchCardModel {
   highlights: SearchResultHighlightType;
   firstIndexed: string;
 
-  constructor(data: any) {
+  constructor(data: any, collectionFilters: AvailableFilterOption[]) {
+    const recordType = getRecordTypeFromURINYPLLink(
+      data.type
+    ) as SearchResultRecordType;
+    // const availableCollectionFilters = filters[]
+    const correspondingCollectionFilter =
+      recordType !== "Item"
+        ? getCollectionFilterFromUUID(data.uuid, collectionFilters)
+        : "";
+    let collectionFilterQueryParam = "";
+    if (recordType === "Collection") {
+      console.log(
+        "correspondingCollectionFilter is: ",
+        correspondingCollectionFilter
+      );
+      collectionFilterQueryParam = correspondingCollectionFilter
+        ? `?filters=[Collection=${correspondingCollectionFilter.name}]`
+        : "";
+      console.log(
+        "collectionFilterQueryParam is: ",
+        collectionFilterQueryParam
+      );
+    }
+
     this.title = data.title;
     this.uuid = data.uuid;
     this.url =
-      data.recordType === "item"
+      recordType === "Item"
         ? `/items/${data.uuid}`
-        : `/collections/${data.uuid}`;
-    this.recordType = capitalize(data.recordType) as SearchResultRecordType;
+        : `/collections/${data.uuid}${collectionFilterQueryParam}`;
```

Comment:
Nit: `data.uuid` is already assigned to `this.uuid`. Pointing it out for consistency.